### ISSUE TITLE
Archeology Digsites removed upon dig attempt

### ISSUE THREAD

**Issue by Vnrs0** [gamefreedomgit/Maelstrom#3638]
[//]: # (REMBEMBER! Add links to things related to the bug using for example:)
[//]: # (http://wowhead.com/)
[//]: # (https://imgur.com/a/BsXR2Tw)

**Archeology Digsites removed upon dig attempt: I suspect it happening when the game can not find a path to the player as it happened on uneven ground or walls**

**Not 100% sure but on the coordinates in the screenshots, I found that if you try to dig on a wall (non-teleportable surface) it gives the error message over and over again. It does, however, not remove the entire digsite like it happens with dwarfen digsites:**

**None of the digsites should be removed unless all 3 artifact spots have been dug up:**

**Database links:**

**Comment by varjla** [comment]
Do you talk about this error, right?

Reaching a digsite, dig it, and it dissapears randomly with an error, generating a new digsite... (I never saw this on other servers)
![WoWScrnShot_061523_100816](https://github.com/gamefreedomgit/Maelstrom/assets/6797090/5d1b7a87-9f91-4d06-b890-76f364fd8aae)

**Comment by Vnrs0** [comment]
> Do you talk about this error, right?
> 
> Reaching a digsite, dig it, and it dissapears randomly with an error, generating a new digsite... (I never saw this on other servers) ![WoWScrnShot_061523_100816](https://user-images.githubusercontent.com/6797090/246039996-5d1b7a87-9f91-4d06-b890-76f364fd8aae.jpg)

Yes, this exact issue. I farmed Archeology for 7 days straight and it did happen a lot of times, mostly on dwarf and troll digsites

**Comment by varjla** [comment]
I just know why this happens. Only to confirm by devs: the archeology spots (not the zone, i talk about the specific item spawn location that you must track with the telescopes) are FIXED on terrain, being only 3/5 spots in each zone. If one of this spots are unreacheable, I guess the system make a check and delete the zone as you dig in that zone. It's my bet for this problem.
